Marlin/src/feature/pause.cpp

```diff
@@ -143,16 +143,18 @@ bool pause_print(const float &retract, const float &z_lift, const float &x_pos,
   // Save current position
   stepper.synchronize();
   COPY(resume_position, current_position);
-
+  set_current_to_destination();            // Reset all old destinations to current
   if (retract) {
     // Initial retract before move to filament change position
     set_destination_to_current();
     destination[E_AXIS] += retract;
     RUNPLAN(PAUSE_PARK_RETRACT_FEEDRATE);
     stepper.synchronize();
+    set_destination_to_current(); // apply actual position value after move
```

Comment:
RUNPLAN Vs Doblockmove
RUNPLAN need ' destinations ' to move and after you manually set ' current pos '
Doblockmoves  apply directly the new current position , and ask planner to go there
For these reason , we are forced to prepare ' current ' and ' destination ' before each 
In this case ' retract is a RUNPLAN , then we ensure the current pos and we set destination and because of a doblockmove just after we have to apply the actual current value '